src/test/scala/mesosphere/mesos/TaskBuilderTest.scala

```diff
@@ -278,6 +278,40 @@ class TaskBuilderTest extends MarathonSpec {
     // TODO test for resources etc.
   }
 
+  test("PortMappingsWithZeroContainerPort") {
+    val offer = makeBasicOfferWithRole(cpus = 1.0, mem = 128.0, disk = 1000.0, beginPort = 31000, endPort = 32000, role = "*")
+      .addResources(ScalarResource("cpus", 1, "*"))
+      .addResources(ScalarResource("mem", 128, "*"))
+      .addResources(ScalarResource("disk", 1000, "*"))
+      .addResources(RangesResource(Resource.PORTS, Seq(protos.Range(33000, 34000)), "marathon"))
+      .build
+
+    val task: Option[(TaskInfo, Seq[Long])] = buildIfMatches(
+      offer, AppDefinition(
+        id = "testApp".toPath,
+        cpus = 1,
+        mem = 64,
+        disk = 1,
+        executor = "//cmd",
+        ports = Seq(8080, 8081),
```

Comment:
you have more ports here then in the port mappings in the container. It would be nice if that matched.

(Actually, we should probably validate that and disallow specifying both, I guess)